src/main/java/com/tobmistaketracker/detector/MistakeDetectors/VerzikMeleeChancedTracker.java

```diff
@@ -0,0 +1,76 @@
+package com.tobmistaketracker.detector.MistakeDetectors;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.Actor;
+import net.runelite.api.NPC;
+import net.runelite.api.Player;
+import net.runelite.api.coords.WorldArea;
+import net.runelite.api.events.AnimationChanged;
+import net.runelite.api.gameval.AnimationID;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@Slf4j
+public class VerzikMeleeChancedTracker {
+    @Getter
+    private final HashSet<Player> playersThatChancedMelee = new HashSet<>();
+
+    private Player lastTickTarget;
+    private WorldArea lastTickTargetArea;
+    private WorldArea lastTickVerzikArea;
+
+    private static final Set<Integer> VERZIK_ATTACK_ANIMATIONS = Set.of(
+            AnimationID.VERZIK_PHASE3_ATTACK_MELEE,
+            AnimationID.VERZIK_PHASE3_ATTACK_MAGIC,
+            AnimationID.VERZIK_PHASE3_ATTACK_RANGED
+    );
+
+    public void setVerzikAttackInfo(NPC verzik){
+        if (verzik == null)
+        {
+            emptyTickData();
+        } else {
+            Actor target = verzik.getInteracting();
+            if (target instanceof Player)
+            {
+                lastTickTarget = (Player) target;
+                lastTickTargetArea = lastTickTarget.getWorldArea();
+                lastTickVerzikArea = verzik.getWorldArea();
```

Comment:
Why do we need these?

Reply:
I meant these two areas. I understand now actually that this is just to see if we were in melee range last tick. Are you sure this works correctly? I typically make use of the raider's `raider.getPreviousWorldLocation()` to get the WorldPoint of where they were in the previous tick (before the update to the next one). If we wanted to still use that, then you could probably take Verzik's current area, increase its radius by 1, add all those tiles to a collection, and see at detection time if the raider's previous location was in those tiles.

Though I've not tested any of this and you have, so I'll defer to your judgement.

That being said, make sure to account for corners. I *believe* she can still melee on a corner tile, and this should account for that.